storage/postgres/tree_storage.go

```diff
@@ -41,7 +41,7 @@ const (
 			FROM subtree n
 			WHERE n.subtree_id IN (` + placeholderSQL + `) AND
 			n.tree_id = ? AND n.subtree_revision <= ?
-			GROUP BY n.subtree_id
+			GROUP BY n.tree_id, n.subtree_id
```

Comment:
Does pgsql have the same optimisation then, or is this speculative?
If you have this in here you might want to update the PR description to mention pgsql too.